### PR TITLE
Support oneOf and anyOf as alternative of enum/enumNames

### PR DESCRIPTION
**Pull request by fbessou**
### Reasons for making this change

As mentioned by @dlax in #532 enumNames keyword is not in the JSON Schema specification (neither draft-04 nor draft-06).
Another way to support choices schemas is to write them with the following structure :
```js
{
  oneOf: [{
    enum: ["foo"],
    title: "Foo"
  }, {
    enum: ["bar"],
    title: "Bar"
  }]
}
```
or with draft-06's new `const` keyword:
```js
{
  oneOf: [{
    const: "foo",
    title: "Foo"
  }, {
    const: "bar",
    title: "Bar"
  }]
}
```
This PR adds basic support for oneOf/anyOf as an alternative to the non standard `enumNames` keywords.

### Checklist

* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated docs if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [x] **I'm adding a new feature**
  - [x] I've updated the playground with an example use of the feature

### CODE REVIEW

File: test/utils_test.js [hunk 1]
```diff
@@ -270,34 +270,59 @@ describe("utils", () => {
...
+        it("should be true if oneOf/anyOf schemas are all constants", () => {
+          const schema = {
+            items: { oneOf: [constantSchema, constantSchema] },
+            uniqueItems: true,
+          };
+          expect(isMultiSelect(schema)).to.be.true;
+        });
+      });
+
+      it("should retrieve reference schema definitions", () => {
```
Comment (glasserc) [inline test/utils_test.js]: `referenced`

File: src/utils.js [hunk 2]
```diff
@@ -294,20 +294,25 @@ export function toConstant(schema) {
...
   }
   return false;
 }
 
+export function isMultiSelect(schema, definitions = {}) {
+  if (!schema.uniqueItems || !schema.items) {
+    return false;
+  }
+  const itemsSchema = retrieveSchema(schema.items, definitions);
+  return isSelect(itemsSchema);
```
Comment (glasserc) [inline src/utils.js]: How about `isSelect(schema.items, definitions)` instead?

**Comment by fbessou** [comment]
Currently there isn't any JSDoc comment in the react-jsonschema-form sources. Should I add a simple comment or a JSDoc comment ?

**Comment by glasserc** [comment]
We don't have any code that does anything specific with JSDoc comments and actually most functions are not documented using JSDoc, but adding a JSDoc comment is fine. It doesn't have to be complicated -- the example on validateFormData is fine.

**Comment by fbessou** [comment]
@glasserc, I made all the requested changes except the renaming of `isSelect` for which I haven't found a better name.

**Comment by glasserc** [comment]
Thanks!

**Comment by glasserc** [comment]
Released in v0.50.0.

**Pull request merged**